crates/lox_core/src/time/continuous.rs

```diff
@@ -1154,260 +1020,362 @@ mod tests {
     }
 
     #[test]
-    fn test_timescale_into_str() {
+    fn test_unscaled_time_days_since_j2000() {
+        struct TestCase {
+            desc: &'static str,
+            time: UnscaledTime,
+            expected: f64,
+        }
+
+        let test_cases = [
+            TestCase {
+                desc: "at the epoch",
+                time: UnscaledTime::default(),
+                expected: 0.0,
+            },
+            TestCase {
+                desc: "exactly one day after the epoch",
+                time: UnscaledTime {
+                    seconds: SECONDS_PER_DAY,
+                    attoseconds: 0,
+                },
+                expected: 1.0,
+            },
+            TestCase {
+                desc: "exactly one day before the epoch",
+                time: UnscaledTime {
+                    seconds: -SECONDS_PER_DAY,
+                    attoseconds: 0,
+                },
+                expected: -1.0,
+            },
+            TestCase {
+                desc: "a partial number of days after the epoch",
+                time: UnscaledTime {
+                    seconds: (SECONDS_PER_DAY / 2) * 3,
+                    attoseconds: ATTOSECONDS_PER_SECOND / 2,
+                },
+                expected: 1.5000057870370371,
+            },
+        ];
+
+        for tc in test_cases {
+            let actual = tc.time.days_since_j2000();
+            assert_float_eq!(
+                tc.expected,
+                actual,
+                abs <= 1e-12,
+                "{}: expected {}, got {}",
+                tc.desc,
+                tc.expected,
+                actual
+            );
+        }
+    }
+
+    #[test]
+    fn test_unscaled_time_centuries_since_j2000() {
```

Comment:
I would recommend the [rstest](https://docs.rs/rstest/latest/rstest/#creating-parametrized-tests) crate for parametrising test.